// voca/helper/format/const.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
// Type specifiers
var TYPE_INTEGER = exports.TYPE_INTEGER = 'i';
var TYPE_INTEGER_BINARY = exports.TYPE_INTEGER_BINARY = 'b';
var TYPE_INTEGER_ASCII_CHARACTER = exports.TYPE_INTEGER_ASCII_CHARACTER = 'c';
var TYPE_INTEGER_DECIMAL = exports.TYPE_INTEGER_DECIMAL = 'd';
var TYPE_INTEGER_OCTAL = exports.TYPE_INTEGER_OCTAL = 'o';
var TYPE_INTEGER_UNSIGNED_DECIMAL = exports.TYPE_INTEGER_UNSIGNED_DECIMAL = 'u';
var TYPE_INTEGER_HEXADECIMAL = exports.TYPE_INTEGER_HEXADECIMAL = 'x';
var TYPE_INTEGER_HEXADECIMAL_UPPERCASE = exports.TYPE_INTEGER_HEXADECIMAL_UPPERCASE = 'X';
var TYPE_FLOAT_SCIENTIFIC = exports.TYPE_FLOAT_SCIENTIFIC = 'e';
var TYPE_FLOAT_SCIENTIFIC_UPPERCASE = exports.TYPE_FLOAT_SCIENTIFIC_UPPERCASE = 'E';
var TYPE_FLOAT = exports.TYPE_FLOAT = 'f';
var TYPE_FLOAT_SHORT = exports.TYPE_FLOAT_SHORT = 'g';
var TYPE_FLOAT_SHORT_UPPERCASE = exports.TYPE_FLOAT_SHORT_UPPERCASE = 'G';
var TYPE_STRING = exports.TYPE_STRING = 's';

// Simple literals
var LITERAL_PERCENT = exports.LITERAL_PERCENT = '%';
var LITERAL_SINGLE_QUOTE = exports.LITERAL_SINGLE_QUOTE = "'";
var LITERAL_PLUS = exports.LITERAL_PLUS = '+';
var LITERAL_MINUS = exports.LITERAL_MINUS = '-';
var LITERAL_PERCENT_SPECIFIER = exports.LITERAL_PERCENT_SPECIFIER = '%%';

// Radix constants to format numbers
var RADIX_BINARY = exports.RADIX_BINARY = 2;
var RADIX_OCTAL = exports.RADIX_OCTAL = 8;
var RADIX_DECIMAL = exports.RADIX_DECIMAL = 10;
var RADIX_HEXADECIMAL = exports.RADIX_HEXADECIMAL = 16;
